airflow/serialization/serialized_objects.py

```diff
@@ -335,6 +334,30 @@ def _value_is_hardcoded_default(cls, attrname: str, value: Any, instance: Any) -
         return False
 
 
+class DependencyDetector:
```

Comment:
Does it have to be a separate class, we just have one method that can be converted to a function, no?

Reply:
I am actually thinking a bit ahead for future extensions:
1) detecting dependencies from dag attributes too (like a list of dependant dag names, etc) -> second method
2) make the class configurable